File: q509.java
class Solution {
    public int fib(int n) {
        int[] f = new int[n+2];
        
        f[0] = 0;
        f[1] = 1;
        for(int i =2; i<=n;i++){
            f[i] = f[i-1]+f[i-2];
        }
        
        return f[n];
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.fib(12));
    }
}

File: q509.rs
fn fib(n: usize) -> usize {
    let mut f = vec![0; n + 2];

    f[0] = 0;
    f[1] = 1;

    let mut i: usize = 2;

    while i <= n {
        f[i] = f[i - 1] + f[i - 2];
        i += 1;
    }

    return f[n];
}

fn main() {
    let res = fib(12);

    println!("{:?}", res);
}